R/ShinyMethods.R

```diff
@@ -305,14 +351,22 @@ server.scfind <- function(object)
             qo.output <- reactive({
                 selected.index <- input$queryOptimizer_rows_selected
                 available.queries <-  recommended.queries()
-                selected.query <- available.queries[selected.index, 'Query']
+                
+                selected.query <- if(all(startsWith(object@index$genes(), "chr") == T)) gsub(":|-", "_", available.queries$Query[selected.index]) else available.queries$Query[selected.index]
                 unlist(strsplit(gsub("\\s", "", selected.query), ","))
             })
 
             output$geneCheckbox <- renderUI({
-                genes.in.list <- as.character(gene.list())
+                genes.in.list <- caseCorrect(object, as.character(gene.list()))
```

Comment:
`gene.list()` returns the gene lists of the input text box. If we refactored this `caseCorrect` to the `gene.list()` we guarantee persistence to avoid wrapping the code and possibly bugs :D